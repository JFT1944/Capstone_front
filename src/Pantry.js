import React, { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import DispensaApi from "./ApiHelper";
import UpdatePanel from "./UpdatePanel";


function Pantry(props){
let {user} = props
let [pantry, setPantry] = useState('Loading...')
let navigate = useNavigate()

useEffect(() => {
try {
    console.log(user)
console.log(typeof(pantry))
DispensaApi.getAllIngredients(user.username).then((res) => {
console.log(res)
setPantry(res)

})
} catch (error) {
    console.log(error)
    navigate('/')
}

}, [])



function handleEditClick(e, x){
    e.preventDefault()
    
    console.log(x)
    navigate(`/pantry/${x.id}`)

}





    return(
        <>
            <div className="ingredient_card_holder">
            {typeof(pantry) === 'string' ? pantry : pantry.map((x) => <div style={x.full_amount === '0' ? {border: '3px solid #AE2C35', background:'rgba(174, 44, 53, .3)'} :  {border: '3px solid #0F683A'}} className="ingredient_card">
                <h3>{x.name}</h3>
                <p>Available: {x.available_amount}{x.unit}</p>
                <p>Total Available: {x.full_amount}{x.unit}</p>
                <button onClick={(e) => handleEditClick(e, x)}>Edit</button>
            </div> )}
            </div>
        </>
    )
}

export default Pantry