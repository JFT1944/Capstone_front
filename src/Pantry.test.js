
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import Pantry from "./Pantry";
import { BrowserRouter, MemoryRouter } from "react-router-dom";


// write a smoke test for the home component

it('should appear in a smoketest', () => {
    <BrowserRouter>
    render(<Pantry user={'user'} token={'token'} setUser={'setUser'}/>)
    </BrowserRouter>
})

it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><Pantry user={'user'} token={'token'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})
