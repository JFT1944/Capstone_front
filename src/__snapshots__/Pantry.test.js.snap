// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ingredient_card_holder"
  >
    Loading...
  </div>
</DocumentFragment>
`;
